src/performance/metrics/StatisticCollector.cpp

```diff
@@ -46,64 +48,172 @@ int StatisticCollector::getMemoryUsageByProcess() {
         }
     }
     fclose(file);
-    std::cout << "Memory Usage: " + std::to_string(result) << std::endl;
+    // std::cout << "Memory Usage: " + std::to_string(result) << std::endl;
     return result;
 }
 
-int StatisticCollector::parseLine(char* line) {
-    int i = strlen(line);
-    const char* p = line;
-    while (*p < '0' || *p > '9') p++;
-    line[i - 3] = '\0';
-    i = atoi(p);
-    return i;
+int StatisticCollector::getThreadCount() {
+    FILE *file = fopen("/proc/self/stat", "r");
+    long result;
+    char line[128];
+
+    for (int i = 0; i < 20; i++) {
```

Comment:
Why is this 20? can we have this in a constant?

Reply:
It's getting the 20th value which contains the thread count

Reply:
Ok